flink-ml-lib/src/main/java/org/apache/flink/ml/classification/naivebayes/NaiveBayesModel.java

```diff
@@ -0,0 +1,226 @@
+        final String featuresCol = getFeaturesCol();
+
+        List<String> colNames = new ArrayList<>(inputs[0].getResolvedSchema().getColumnNames());
+        colNames.add(predictionCol);
+
+        List<TypeInformation<?>> colTypes =
+                new ArrayList<>(inputs[0].getResolvedSchema().getColumnDataTypes())
+                        .stream()
+                                .map(
+                                        (Function<DataType, TypeInformation<?>>)
+                                                dataType ->
+                                                        TypeInformation.of(
+                                                                dataType.getConversionClass()))
+                                .collect(Collectors.toList());
+        colTypes.add(TypeInformation.of(Object.class));
+
+        StreamTableEnvironment tEnv =
+                (StreamTableEnvironment) ((TableImpl) modelTable).getTableEnvironment();
+        DataStream<NaiveBayesModelData> modelStream =
+                NaiveBayesModelData.toDataStream(tEnv, modelTable);
+        DataStream<Row> input = tEnv.toDataStream(inputs[0]);
+
+        Map<String, DataStream<?>> broadcastMap = new HashMap<>();
+        broadcastMap.put(broadcastModelKey, modelStream);
+
+        Function<List<DataStream<?>>, DataStream<Row>> function =
+                dataStreams -> {
+                    DataStream stream = dataStreams.get(0);
+                    return stream.transform(
+                            this.getClass().getSimpleName(),
+                            new RowTypeInfo(
+                                    colTypes.toArray(new TypeInformation[0]),
+                                    colNames.toArray(new String[0])),
+                            new NaiveBayesPredictOp(new NaiveBayesPredictFunc(featuresCol)));
+                };
+        DataStream<Row> output =
+                BroadcastUtils.withBroadcastStream(
+                        Collections.singletonList(input), broadcastMap, function);
+
+        Table outputTable = tEnv.fromDataStream(output);
+
+        return new Table[] {outputTable};
+    }
+
+    @Override
+    public void save(String path) throws IOException {
+        StreamTableEnvironment tEnv =
+                (StreamTableEnvironment) ((TableImpl) modelTable).getTableEnvironment();
+
+        String dataPath = ReadWriteUtils.getDataPath(path);
+        FileSink<NaiveBayesModelData> sink =
+                FileSink.forRowFormat(
+                                new Path(dataPath), new NaiveBayesModelData.ModelDataEncoder())
+                        .withRollingPolicy(OnCheckpointRollingPolicy.build())
+                        .withBucketAssigner(new BasePathBucketAssigner<>())
+                        .build();
+        NaiveBayesModelData.toDataStream(tEnv, modelTable).sinkTo(sink);
+
+        ReadWriteUtils.saveMetadata(this, path);
+    }
+
+    public static NaiveBayesModel load(String path) throws IOException {
```

Comment:
Could you follow the example of KMeans and the Java doc in `Stage.java` to pass env as argument?